### PR TITLE
Add option for 'GLOBAL_MANAGED_PROXY' proxy_only subnets

### PR DESCRIPTION
**Pull request by LucaPrete**
Allow arbitrary purpose for proxy only subnets.

### CODE REVIEW

File: modules/net-vpc/subnets.tf [hunk 1]
```diff
@@ -134,7 +134,7 @@ resource "google_compute_subnetwork" "proxy_only" {
   region        = each.value.region
   name          = each.value.name
   ip_cidr_range = each.value.ip_cidr_range
-  purpose       = "REGIONAL_MANAGED_PROXY"
```
Comment (LucaPrete) [inline modules/net-vpc/subnets.tf:137]: @averbuks @ludoo @juliocc as an alternative, we can also define two different blocks:
1) proxy_only_regional
2) proxy_only_global

...So we remain consistent with the rest of the code.

wdyt?

Comment (juliocc) [reply modules/net-vpc/subnets.tf:137]: I like a single block with the user specifying the purpose.

**Review by juliocc — APPROVED** [review]

File: modules/net-vpc/subnets.tf (hunk 1 above)
Comment (averbuks) [reply modules/net-vpc/subnets.tf:137]: +1, I would even stick to just `subnets` var in the future if that comes to the mainstream https://www.terraform.io/language/functions/defaults (still experimental)

**Review by averbuks — APPROVED** [review]

**Comment by ludoo** [comment]
Is this still being worked on, or shall we close it?

**Pull request closed without merging**